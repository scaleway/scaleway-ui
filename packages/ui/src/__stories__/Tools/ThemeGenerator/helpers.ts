// Function imported from https://github.com/PimpTrizkit/PJs/wiki/12.-Shade,-Blend-and-Convert-a-Web-Color-(pSBC.js)#stackoverflow-archive-begin to shade hexa colors
type ShadeHexColorType = (color: string, percent: number) => string

export const shadeHexColor: ShadeHexColorType = (color, percent) => {
  const f = parseInt(color.slice(1), 16)
  const t = percent < 0 ? 0 : 255
  const p = percent < 0 ? percent * -1 : percent
  // eslint-disable-next-line no-bitwise
  const R = f >> 16
  // eslint-disable-next-line no-bitwise
  const G = (f >> 8) & 0x00ff
  // eslint-disable-next-line no-bitwise
  const B = f & 0x0000ff

  return `#${(
    0x1000000 +
    (Math.round((t - R) * p) + R) * 0x10000 +
    (Math.round((t - G) * p) + G) * 0x100 +
    (Math.round((t - B) * p) + B)
  )
    .toString(16)
    .slice(1)}`
}

export const generateShadeContrast = (
  shadeKey: string,
  value: string,
  index: number,
) =>
  Number(shadeKey) < 900
    ? shadeHexColor(value, (1 / 15) * (index + 1))
    : shadeHexColor(value, -(1 / 15) * (index + 1))
