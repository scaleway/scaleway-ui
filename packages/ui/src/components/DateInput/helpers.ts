import { CURRENT_MONTH, CURRENT_YEAR } from './constants'

// First day of the month for a given year
export const getMonthFirstDay = (
  month = CURRENT_MONTH,
  year = CURRENT_YEAR,
) => {
  const firstDay = new Date(year, month - 1, 1).getDay()

  // Change so that a week starts on monday
  return firstDay === 0 ? 6 : firstDay - 1
}

export const addZero = (value: number) => `${value}`.padStart(2, '0')

export const getPreviousMonth = (month: number, year: number) => {
  if (month === 1) {
    return [12, year - 1]
  }

  return [month - 1, year]
}

export const getNextMonth = (month: number, year: number) => {
  if (month === 12) {
    return [1, year + 1]
  }

  return [month + 1, year]
}

// Checks if two date values are of the same month and year
export const isSameMonth = (date: Date, basedate = new Date()) =>
  basedate.getMonth() === date.getMonth() &&
  basedate.getFullYear() === date.getFullYear()

// (bool) Checks if two date values are the same day
export const isSameDay = (date: Date, basedate = new Date()) =>
  basedate.getDate() === date.getDate() &&
  basedate.getMonth() + 1 === date.getMonth() + 1 &&
  basedate.getFullYear() === date.getFullYear()

// Default format if none is provided
const getDateISO = (showMonthYearPicker: boolean, date?: Date) => {
  if (date) {
    if (showMonthYearPicker) {
      return [addZero(date.getMonth() + 1), date.getFullYear()].join('/')
    }

    return [
      addZero(date.getMonth() + 1),
      addZero(date.getDate()),
      date.getFullYear(),
    ].join('/')
  }

  return ''
}

export const formatValue = (
  computedValue: Date | null,
  computedRange: {
    start: Date | null
    end: Date | null
  } | null,
  showMonthYearPicker: boolean,
  selectsRange: boolean,
  format?: (value?: Date) => string | undefined,
) => {
  if (selectsRange && computedRange) {
    return format
      ? `${format(computedRange.start ?? undefined) ? `${format(computedRange.start ?? undefined)} - ` : ''}${format(computedRange.end ?? undefined) ?? ''}`
      : `${getDateISO(showMonthYearPicker, computedRange.start ?? undefined)}${computedRange.start ? ' - ' : ''}${getDateISO(showMonthYearPicker, computedRange.end ?? undefined)}`
  }

  if (computedValue && format) {
    return format(computedValue)
  }
  if (computedValue) {
    return getDateISO(showMonthYearPicker, computedValue)
  }

  return undefined
}
