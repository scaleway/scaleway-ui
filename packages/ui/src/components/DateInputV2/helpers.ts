export const POPUP_WIDTH = '264px'

// The current year
export const CURRENT_YEAR = new Date().getFullYear()

// The current month starting from 1 - 12
// 1 => January, 12 => December
export const CURRENT_MONTH = new Date().getMonth() + 1

// The current day 1-28/29/30/31

// Weeks displayed on calendar
export const CALENDAR_WEEKS = 6

// Number days in a month for a given year
export const getMonthDays = (month = CURRENT_MONTH, year = CURRENT_YEAR) => {
  const months30 = [4, 6, 9, 11] // 30 days in april, june, september and november
  const leapYear = year % 4 === 0

  if (month === 2) {
    return leapYear ? 29 : 28 // Days in february
  }

  return months30.includes(month) ? 30 : 31
}

// First day of the month for a given year from 1 (monday) - 7 (saturday)
export const getMonthFirstDay = (month = CURRENT_MONTH, year = CURRENT_YEAR) =>
  new Date(`${year}-${month}-01`).getDay() + 1

// Adds a zero for 1-digit months/days
export const addZero = (value: number) => `${value}`.padStart(2, '0')

export const getPreviousMonth = (month: number, year: number) => {
  if (month === 1) {
    return [12, year - 1]
  }

  return [month - 1, year]
}

export const getNextMonth = (month: number, year: number) => {
  if (month === 12) {
    return [1, year + 1]
  }

  return [month + 1, year]
}

// Checks if two date values are of the same month and year
export const isSameMonth = (date: Date, basedate = new Date()) =>
  basedate.getMonth() + 1 === date.getMonth() + 1 &&
  basedate.getFullYear() === date.getFullYear()

// (bool) Checks if two date values are the same day
export const isSameDay = (date: Date, basedate = new Date()) =>
  basedate.getDate() === date.getDate() &&
  basedate.getMonth() + 1 === date.getMonth() + 1 &&
  basedate.getFullYear() === date.getFullYear()
