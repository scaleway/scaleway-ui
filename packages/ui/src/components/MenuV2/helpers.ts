import { Children, type ReactNode, isValidElement } from 'react'

/**
 * Search inside a children (React Element) recursively until a result is found
 */
export const searchChildren = (
  children: ReactNode,
  searchString: string,
): ReactNode[] => {
  const matches: ReactNode[] = []
  const searchRegex = new RegExp(searchString, 'i')

  Children.forEach(children, child => {
    if (typeof child === 'string' && child.match(searchRegex)) {
      matches.push(child)
    } else if (isValidElement(child)) {
      const childProps = child.props as { children: ReactNode }

      // Recursively search the children of this element
      const childMatches = searchChildren(childProps.children, searchString)

      if (childMatches.length > 0) {
        // If any matches are found within this child's children, push the entire child element
        matches.push(child)
      }
    }
  })

  return matches
}
