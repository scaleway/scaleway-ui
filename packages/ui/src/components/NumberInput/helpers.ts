export const bounded = (value: number, min: number, max: number) =>
  Math.max(min, Math.min(value, max))

export const roundStep = (value: number, step: number, direction: number) =>
  direction === -1
    ? Math.floor(value / step) * step
    : Math.ceil(value / step) * step

export const getMinusRoundedValue = (currentValue: number, step: number) =>
  currentValue % step === 0
    ? roundStep(currentValue - step, step, -1)
    : roundStep(currentValue, step, -1)

export const getPlusRoundedValue = (currentValue: number, step: number) =>
  currentValue % step === 0
    ? roundStep(currentValue + step, step, 1)
    : roundStep(currentValue, step, 1)
