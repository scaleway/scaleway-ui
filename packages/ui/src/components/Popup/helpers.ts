import type { RefObject } from 'react'

export type PopupPlacement = 'top' | 'right' | 'bottom' | 'left' | 'auto'
export const ARROW_WIDTH = 8 // in px
const SPACE = 4 // in px
const TOTAL_USED_SPACE = 0 // in px
export const DEFAULT_POSITIONS = {
  arrowLeft: -999,
  arrowTop: -999,
  arrowTransform: 'translate(-50%, -50)',
  placement: 'top',
  rotate: 135,
  popupInitialPosition: 'translate3d(-999px, -999px, 0)',
  popupPosition: 'translate3d(-999px, -999px, 0)',
}

type ComputePlacementTypes = {
  childrenStructuredRef: DOMRect
  popupStructuredRef: DOMRect
  popupPortalTarget: HTMLElement
  offsetParentRect: DOMRect
  offsetParent: Element
}

/**
 * This function will find the best placement in a window for popup based on children position and popup size
 */
const computePlacement = ({
  childrenStructuredRef,
  popupStructuredRef,
  offsetParentRect,
  offsetParent,
  popupPortalTarget,
}: ComputePlacementTypes) => {
  const {
    top: childrenTop,
    left: childrenLeft,
    right: childrenRight,
  } = childrenStructuredRef

  const {
    top: parentTop,
    left: parentLeft,
    right: parentRight,
  } = offsetParentRect

  const isPopupPortalTargetBody =
    popupPortalTarget === document.body || offsetParent === document.body

  const overloadedChildrenLeft = isPopupPortalTargetBody
    ? childrenLeft
    : childrenLeft - parentLeft
  const overloadedChildrenTop = isPopupPortalTargetBody
    ? childrenTop
    : childrenTop - parentTop
  const overloadedChildrenRight = isPopupPortalTargetBody
    ? childrenRight
    : childrenRight - parentRight

  const { width: popupWidth, height: popupHeight } = popupStructuredRef

  if (overloadedChildrenTop - popupHeight - TOTAL_USED_SPACE < 0) {
    return 'bottom'
  }

  if (overloadedChildrenLeft - popupWidth - TOTAL_USED_SPACE < 0) {
    return 'right'
  }

  if (
    overloadedChildrenRight + popupWidth + TOTAL_USED_SPACE >
    window.innerWidth
  ) {
    return 'left'
  }

  return 'top'
}

/**
 * This function will check if the offset parent is usable for popup positioning
 * If not it will loop and search for a compatible parent until document.body is reached
 */
const findOffsetParent = (element: RefObject<HTMLDivElement>) => {
  const offsetParent = element?.current?.offsetParent

  // We need to check if offsetParent is a table cell or a table because they are not suitable for positioning
  // https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/offsetParent
  if (offsetParent && !['TH', 'TD', 'TABLE'].includes(offsetParent.tagName)) {
    return offsetParent
  }

  let currentElement = element?.current

  while (currentElement && currentElement.tagName !== 'BODY') {
    const { position } = window.getComputedStyle(currentElement)

    // Check if the current element is a potential offset parent
    if (position !== 'static') {
      return currentElement
    }

    currentElement = currentElement.parentElement as HTMLDivElement
  }

  // If no suitable offset parent is found, return the body element
  return document.body
}

type ComputePositionsTypes = {
  placement: PopupPlacement
  childrenRef: RefObject<HTMLDivElement>
  popupRef: RefObject<HTMLDivElement>
  popupPortalTarget: HTMLElement
}

/**
 * This function will compute the positions of popup and arrow based on children position and popup size
 */
export const computePositions = ({
  placement,
  childrenRef,
  popupRef,
  popupPortalTarget,
}: ComputePositionsTypes) => {
  const childrenRect = (
    childrenRef.current as HTMLDivElement
  ).getBoundingClientRect()
  const offsetParent = findOffsetParent(childrenRef)
  const offsetParentRect = offsetParent?.getBoundingClientRect() ?? {
    top: 0,
    left: 0,
    right: 0,
  }
  const popupStructuredRef = (
    popupRef.current as HTMLDivElement
  ).getBoundingClientRect()

  const placementBasedOnWindowSize =
    placement === 'auto'
      ? computePlacement({
          childrenStructuredRef: childrenRect,
          popupStructuredRef,
          offsetParentRect,
          popupPortalTarget,
          offsetParent,
        })
      : placement

  const {
    top: childrenTop,
    left: childrenLeft,
    right: childrenRight,
    width: childrenWidth,
    height: childrenHeight,
  } = childrenRect

  const {
    top: parentTop,
    left: parentLeft,
    right: parentRight,
  } = offsetParentRect

  const { width: popupWidth, height: popupHeight } = popupStructuredRef

  // offSetParent is the closest positioned ancestor. If the element is not positioned, the nearest table cell or root element is used.
  const isPopupPortalTargetBody =
    popupPortalTarget === document.body || offsetParent === document.body

  // It will get how much scroll is done on the page to compute the position of the popup
  const scrollTopValue = isPopupPortalTargetBody
    ? document.documentElement.scrollTop
    : offsetParent.scrollTop

  // We need to compute the position of the popup based on the parent element in the case the popup is not in the body
  const overloadedChildrenLeft = isPopupPortalTargetBody
    ? childrenLeft
    : childrenLeft - parentLeft
  const overloadedChildrenTop = isPopupPortalTargetBody
    ? childrenTop
    : childrenTop - parentTop
  const overloadedChildrenRight = isPopupPortalTargetBody
    ? childrenRight
    : childrenRight + childrenWidth + ARROW_WIDTH + SPACE - parentRight / 2

  switch (placementBasedOnWindowSize) {
    case 'bottom': {
      const positionX =
        overloadedChildrenLeft + childrenWidth / 2 - popupWidth / 2
      const positionY =
        overloadedChildrenTop +
        scrollTopValue +
        childrenHeight +
        ARROW_WIDTH +
        SPACE

      return {
        arrowLeft: popupWidth / 2,
        arrowTop: -ARROW_WIDTH - 5,
        arrowTransform: '',
        placement: 'bottom',
        rotate: 180,
        popupInitialPosition: `translate3d(${positionX}px, ${
          positionY - TOTAL_USED_SPACE
        }px, 0)`,
        popupPosition: `translate3d(${positionX}px, ${positionY}px, 0)`,
      }
    }
    case 'left': {
      const positionX =
        overloadedChildrenLeft - popupWidth - ARROW_WIDTH - SPACE * 2
      const positionY =
        overloadedChildrenTop +
        scrollTopValue -
        popupHeight / 2 +
        childrenHeight / 2

      return {
        arrowLeft: popupWidth + ARROW_WIDTH + 5,
        arrowTop: popupHeight / 2,
        arrowTransform: 'translate(-50%, -50%)',
        placement: 'left',
        rotate: -90,
        popupInitialPosition: `translate3d(${
          positionX + TOTAL_USED_SPACE
        }px, ${positionY}px, 0)`,
        popupPosition: `translate3d(${positionX}px, ${positionY}px, 0)`,
      }
    }
    case 'right': {
      const positionX = overloadedChildrenRight + ARROW_WIDTH + SPACE * 2
      const positionY =
        overloadedChildrenTop +
        scrollTopValue -
        popupHeight / 2 +
        childrenHeight / 2

      return {
        arrowLeft: -ARROW_WIDTH - 5,
        arrowTop: popupHeight / 2,
        arrowTransform: 'translate(50%, -50%)',
        placement: 'right',
        rotate: 90,
        popupInitialPosition: `translate3d(${
          positionX - TOTAL_USED_SPACE
        }px, ${positionY}px, 0)`,
        popupPosition: `translate3d(${positionX}px, ${positionY}px, 0)`,
      }
    }
    default: {
      // top placement is default value
      const positionX =
        overloadedChildrenLeft + childrenWidth / 2 - popupWidth / 2
      const positionY =
        overloadedChildrenTop +
        scrollTopValue -
        popupHeight -
        ARROW_WIDTH -
        SPACE

      return {
        arrowLeft: popupWidth / 2,
        arrowTop: popupHeight - 1,
        arrowTransform: '',
        placement: 'top',
        rotate: 0,
        popupInitialPosition: `translate3d(${positionX}px, ${
          positionY + TOTAL_USED_SPACE
        }px, 0)`,
        popupPosition: `translate3d(${positionX}px, ${positionY}px, 0)`,
      }
    }
  }
}
