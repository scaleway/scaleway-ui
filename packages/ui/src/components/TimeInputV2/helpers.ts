const format = (value: string, timeFormat: 12 | 24, isHour?: boolean) => {
  if (value === '') return ''
  if (timeFormat === 12 && isHour) {
    const formattedValue = Number.parseInt(value, 10) % 12
    const finalValue = formattedValue === 0 ? 12 : formattedValue

    return finalValue.toString().padStart(2, '0')
  }

  return value.padStart(2, '0')
}

const computePeriod = (hours?: string, period?: 'am' | 'pm') => {
  if (period) return period

  if (hours && Number.parseInt(hours, 10) >= 12) return 'pm'

  return 'am'
}

export const isNumber = (key: string) => /^\d+$/.test(key)
export const isAOrP = (key: string) => ['a', 'A', 'p', 'P'].includes(key)

const isValidNewHour = (old: string, key: string, timeFormat: number) => {
  const newValue = Number.parseInt(old + key, 10)

  return (
    (timeFormat === 24 && newValue < 24) || (timeFormat === 12 && newValue < 13)
  )
}

export const isCompleteHour = (timeFormat: number, value?: string) => {
  const computedValue = Number.parseInt(value ?? '0', 10)

  if (timeFormat === 12 && computedValue > 1) return true
  if (timeFormat === 24 && computedValue > 2) return true

  return false
}

export const canConcat = (
  oldValue: string | undefined,
  type: 'h' | 'm' | 's',
  key: string,
  timeFormat: 24 | 12,
) =>
  oldValue &&
  ((type === 'h' && oldValue && isValidNewHour(oldValue, key, timeFormat)) ||
    (type !== 'h' && Number.parseInt(oldValue, 10) < 6))

export const formatValue = (
  value: {
    h: string
    m: string
    s: string
    period?: string
  },
  timeFormat: 24 | 12,
) => ({
  h: format(value.h, timeFormat, true),
  m: format(value.m, timeFormat),
  s: format(value.s, timeFormat),
  period: computePeriod(
    value.h,
    value.period === 'am' || value.period === 'pm' ? value.period : undefined,
  ),
})
