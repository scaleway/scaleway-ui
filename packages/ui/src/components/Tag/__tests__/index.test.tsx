import { describe, test } from '@jest/globals'
import { Tag } from '..'
import { shouldMatchEmotionSnapshot } from '../../../../.jest/helpers'

describe('Tag', () => {
  test('renders correctly', () => shouldMatchEmotionSnapshot(<Tag>test</Tag>))

  test('renders correctly neutral', () =>
    shouldMatchEmotionSnapshot(<Tag sentiment="neutral">test</Tag>))

  test('renders correctly disabled', () =>
    shouldMatchEmotionSnapshot(<Tag disabled>test</Tag>))

  test('renders correctly colored', () =>
    shouldMatchEmotionSnapshot(<Tag sentiment="primary">test</Tag>))

  test('renders correctly with icon', () =>
    shouldMatchEmotionSnapshot(<Tag icon="lock">test</Tag>))

  test('renders correctly with isLoading', () =>
    shouldMatchEmotionSnapshot(<Tag isLoading>test</Tag>))

  test('renders correctly with onClose', () =>
    shouldMatchEmotionSnapshot(<Tag onClose={() => {}}>test</Tag>))

  test('renders correctly with copiable', () =>
    shouldMatchEmotionSnapshot(<Tag copiable>test</Tag>))
})
