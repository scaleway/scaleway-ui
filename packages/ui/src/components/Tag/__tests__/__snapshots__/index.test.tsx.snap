// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag renders correctly 1`] = `
<DocumentFragment>
  .cache-1anauza-StyledContainer {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  white-space: nowrap;
  border-radius: 4px;
  padding: 0 8px;
  gap: 8px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  height: 24px;
  color: #3f4250;
  background: #ffffff;
  border: solid 1px #d9dadd;
}

.cache-t6vuyb-StyledTag {
  font-size: 12px;
  font-weight: 500;
  color: inherit;
  max-width: 232px;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

<span
    class="cache-1anauza-StyledContainer el6733p2"
  >
    <span
      aria-disabled="false"
      class="cache-t6vuyb-StyledTag el6733p1"
    >
      test
    </span>
  </span>
</DocumentFragment>
`;

exports[`Tag renders correctly colored 1`] = `
<DocumentFragment>
  .cache-1yta84r-StyledContainer {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  white-space: nowrap;
  border-radius: 4px;
  padding: 0 8px;
  gap: 8px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  height: 24px;
  color: #521094;
  background: #f1eefc;
  border: solid 1px #f1eefc;
}

.cache-t6vuyb-StyledTag {
  font-size: 12px;
  font-weight: 500;
  color: inherit;
  max-width: 232px;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

<span
    class="cache-1yta84r-StyledContainer el6733p2"
  >
    <span
      aria-disabled="false"
      class="cache-t6vuyb-StyledTag el6733p1"
    >
      test
    </span>
  </span>
</DocumentFragment>
`;

exports[`Tag renders correctly disabled 1`] = `
<DocumentFragment>
  .cache-4ijz8s-StyledContainer {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  white-space: nowrap;
  border-radius: 4px;
  padding: 0 8px;
  gap: 8px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  height: 24px;
  color: #b5b7bd;
  background: #f3f3f4;
  border: solid 1px #e9eaeb;
  cursor: not-allowed;
}

.cache-t6vuyb-StyledTag {
  font-size: 12px;
  font-weight: 500;
  color: inherit;
  max-width: 232px;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

<span
    class="cache-4ijz8s-StyledContainer el6733p2"
  >
    <span
      aria-disabled="true"
      class="cache-t6vuyb-StyledTag el6733p1"
    >
      test
    </span>
  </span>
</DocumentFragment>
`;

exports[`Tag renders correctly neutral 1`] = `
<DocumentFragment>
  .cache-1anauza-StyledContainer {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  white-space: nowrap;
  border-radius: 4px;
  padding: 0 8px;
  gap: 8px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  height: 24px;
  color: #3f4250;
  background: #ffffff;
  border: solid 1px #d9dadd;
}

.cache-t6vuyb-StyledTag {
  font-size: 12px;
  font-weight: 500;
  color: inherit;
  max-width: 232px;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

<span
    class="cache-1anauza-StyledContainer el6733p2"
  >
    <span
      aria-disabled="false"
      class="cache-t6vuyb-StyledTag el6733p1"
    >
      test
    </span>
  </span>
</DocumentFragment>
`;

exports[`Tag renders correctly with copiable 1`] = `
<DocumentFragment>
  .cache-1hzk1lh-StyledChildrenContainer {
  display: inherit;
}

.cache-1hzk1lh-StyledChildrenContainer[data-container-full-width='true'] {
  width: 100%;
}

.cache-1ykqsf4-Container-StyledContainer {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  white-space: nowrap;
  border-radius: 4px;
  padding: 0 8px;
  gap: 8px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  height: 24px;
  color: #3f4250;
  background: #ffffff;
  border: solid 1px #d9dadd;
}

.cache-1ykqsf4-Container-StyledContainer:hover,
.cache-1ykqsf4-Container-StyledContainer:active {
  cursor: pointer;
  background: #e9eaeb;
  border-color: #92959d;
}

.cache-1ykqsf4-Container-StyledContainer:active {
  box-shadow: 0px 0px 0px 3px #151A2D5C;
}

.cache-t6vuyb-StyledTag {
  font-size: 12px;
  font-weight: 500;
  color: inherit;
  max-width: 232px;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

<div
    aria-describedby=":r1:"
    class="cache-1hzk1lh-StyledChildrenContainer efkq4700"
    tabindex="0"
  >
    <button
      class="cache-1ykqsf4-Container-StyledContainer el6733p3"
    >
      <span
        aria-disabled="false"
        class="cache-t6vuyb-StyledTag el6733p1"
      >
        test
      </span>
    </button>
  </div>
</DocumentFragment>
`;

exports[`Tag renders correctly with icon 1`] = `
<DocumentFragment>
  .cache-1anauza-StyledContainer {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  white-space: nowrap;
  border-radius: 4px;
  padding: 0 8px;
  gap: 8px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  height: 24px;
  color: #3f4250;
  background: #ffffff;
  border: solid 1px #d9dadd;
}

.cache-1ah7xw8-sizeStyles {
  vertical-align: middle;
  fill: currentColor;
  height: 16px;
  width: 16px;
  min-width: 16px;
  min-height: 16px;
}

.cache-t6vuyb-StyledTag {
  font-size: 12px;
  font-weight: 500;
  color: inherit;
  max-width: 232px;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

<span
    class="cache-1anauza-StyledContainer el6733p2"
  >
    <svg
      class="cache-1ah7xw8-sizeStyles e1gt4cfo0"
      viewBox="0 0 24 24"
    >
      <path
        d="M12,17A2,2 0 0,0 14,15C14,13.89 13.1,13 12,13A2,2 0 0,0 10,15A2,2 0 0,0 12,17M18,8A2,2 0 0,1 20,10V20A2,2 0 0,1 18,22H6A2,2 0 0,1 4,20V10C4,8.89 4.9,8 6,8H7V6A5,5 0 0,1 12,1A5,5 0 0,1 17,6V8H18M12,3A3,3 0 0,0 9,6V8H15V6A3,3 0 0,0 12,3Z"
      />
    </svg>
    <span
      aria-disabled="false"
      class="cache-t6vuyb-StyledTag el6733p1"
    >
      test
    </span>
  </span>
</DocumentFragment>
`;

exports[`Tag renders correctly with isLoading 1`] = `
<DocumentFragment>
  .cache-1anauza-StyledContainer {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  white-space: nowrap;
  border-radius: 4px;
  padding: 0 8px;
  gap: 8px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  height: 24px;
  color: #3f4250;
  background: #ffffff;
  border: solid 1px #d9dadd;
}

.cache-t6vuyb-StyledTag {
  font-size: 12px;
  font-weight: 500;
  color: inherit;
  max-width: 232px;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

.cache-cmn72i-StyledSvg {
  -webkit-animation: spin 0.75s linear infinite;
  animation: spin 0.75s linear infinite;
}

.cache-11f05l0-Loader {
  -webkit-transition: stroke-dashoffset 0.5s ease 0s;
  transition: stroke-dashoffset 0.5s ease 0s;
}

<span
    class="cache-1anauza-StyledContainer el6733p2"
  >
    <span
      aria-disabled="false"
      class="cache-t6vuyb-StyledTag el6733p1"
    >
      test
    </span>
    <svg
      aria-label="Loading"
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="20"
      aria-valuetext="20%"
      class="cache-cmn72i-StyledSvg e1y1n78x1"
      role="progressbar"
      style="height: 16px; width: 16px;"
      viewBox="0 0 100 100"
    >
      <circle
        cx="50"
        cy="50"
        fill="none"
        r="42"
        stroke="#e9eaeb"
        stroke-width="16"
      />
      <circle
        class="cache-11f05l0-Loader"
        cx="50"
        cy="50"
        fill="none"
        r="42"
        stroke="#521094"
        stroke-dasharray="263.89378290154264"
        stroke-dashoffset="211.11502632123413"
        stroke-linecap="round"
        stroke-width="16"
      />
    </svg>
  </span>
</DocumentFragment>
`;

exports[`Tag renders correctly with onClose 1`] = `
<DocumentFragment>
  .cache-1anauza-StyledContainer {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  white-space: nowrap;
  border-radius: 4px;
  padding: 0 8px;
  gap: 8px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  height: 24px;
  color: #3f4250;
  background: #ffffff;
  border: solid 1px #d9dadd;
}

.cache-t6vuyb-StyledTag {
  font-size: 12px;
  font-weight: 500;
  color: inherit;
  max-width: 232px;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

.cache-173kxau-StyledGhostButton-StyledCloseButton {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 32px;
  padding: 0 8px;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 8px;
  border-radius: 4px;
  box-sizing: border-box;
  width: 32px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  outline-offset: 2px;
  white-space: nowrap;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 14px;
  font-family: Asap;
  font-weight: 500;
  letter-spacing: 0;
  line-height: 20px;
  paragraph-spacing: 0;
  text-case: none;
  background: none;
  border: none;
  color: #3f4250;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  padding: 2px;
}

.cache-173kxau-StyledGhostButton-StyledCloseButton:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.cache-173kxau-StyledGhostButton-StyledCloseButton:active {
  box-shadow: 0px 0px 0px 3px #151A2D5C;
}

.cache-173kxau-StyledGhostButton-StyledCloseButton:hover,
.cache-173kxau-StyledGhostButton-StyledCloseButton:active {
  background: #e9eaeb;
  color: #222638;
}

.cache-1ah7xw8-sizeStyles {
  vertical-align: middle;
  fill: currentColor;
  height: 16px;
  width: 16px;
  min-width: 16px;
  min-height: 16px;
}

<span
    class="cache-1anauza-StyledContainer el6733p2"
  >
    <span
      aria-disabled="false"
      class="cache-t6vuyb-StyledTag el6733p1"
    >
      test
    </span>
    <button
      aria-label="Close tag"
      class="el6733p0 cache-173kxau-StyledGhostButton-StyledCloseButton e112qvla0"
      data-testid="close-tag"
      type="button"
    >
      <svg
        class="cache-1ah7xw8-sizeStyles e1gt4cfo0"
        viewBox="0 0 24 24"
      >
        <path
          d="M19,6.41L17.59,5L12,10.59L6.41,5L5,6.41L10.59,12L5,17.59L6.41,19L12,13.41L17.59,19L19,17.59L13.41,12L19,6.41Z"
        />
      </svg>
    </button>
  </span>
</DocumentFragment>
`;
